M2Mqtt/MqttClient.cs

```diff
@@ -663,32 +663,67 @@ public ushort Unsubscribe(string[] topics)
         }
 
         /// <summary>
-        /// Publish a message asynchronously (QoS Level 0 and not retained)
-        /// </summary>
-        /// <param name="topic">Message topic</param>
-        /// <param name="message">Message data (payload)</param>
-        /// <returns>Message Id related to PUBLISH message</returns>
-        public ushort Publish(string topic, byte[] message)
+        /// Publish a message asynchronously (QoS Level <see cref="MqttQoSLevel.AtMostOnce"/> and not retained).
+        /// </summary>
+        /// <param name="topic">Message topic.</param>
+        /// <param name="message">Message data (payload).</param>
+        /// <returns>Message Id related to PUBLISH message.</returns>
+        public ushort Publish(
+            string topic,
+            byte[] message)
         {
-            return Publish(topic, message, MqttQoSLevel.AtMostOnce, false);
+            return Publish(
+                topic,
+                message,
+                null,
+                MqttQoSLevel.AtMostOnce,
+                false);
         }
 
         /// <summary>
-        /// Publish a message asynchronously
+        /// Publish a message asynchronously (QoS Level <see cref="MqttQoSLevel.AtMostOnce"/> and not retained).
         /// </summary>
-        /// <param name="topic">Message topic</param>
-        /// <param name="message">Message data (payload)</param>
-        /// <param name="qosLevel">QoS Level</param>
-        /// <param name="retain">Retain flag</param>
-        /// <returns>Message Id related to PUBLISH message</returns>
-        public ushort Publish(string topic, byte[] message, MqttQoSLevel qosLevel, bool retain)
+        /// <param name="topic">Message topic.</param>
+        /// <param name="message">Message data (payload).</param>
+        /// <param name="contentType">Content of the application message.</param>
```

Comment:
This is supported only for MQTT5. There is no notion of Content Type in MQTT 3.1.1. See:
https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901065 (and it is not present in the MQTT3.1.1 spec http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html)

So please mention this is only for MQTT5 and does not apply to any other version. I don't see any reason why we *should* check that the MQTT level is the proper one but at least the description should be clear

Reply:
Yes, I've checked that and also Microsoft docs on the Azure support for it [here](https://learn.microsoft.com/azure/iot-hub/iot-hub-mqtt-5?WT.mc_id=WT.mc_id%3DDT-MVP-5004179) and [here](https://learn.microsoft.com/azure/iot-hub/iot-hub-mqtt-5-reference?WT.mc_id=WT.mc_id%3DDT-MVP-5004179).

Just added that to the comments to make it clear. As for validation all this it's already handled properly when composing the binary payload so I don't think we need to add any overhead here with checks... I mean it won't break anything if a developer sets this on a v3 client.